api/src/handlers.rs

```diff
@@ -405,8 +400,15 @@ pub async fn get_leaderboard(
         }));
     }
 
+    let limit = if project_ids.is_empty() {
```

Comment:
Let's rename 'limit' to something else as it's a bit confusing since we already have a param called limit.